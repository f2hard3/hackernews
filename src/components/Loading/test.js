import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import { ButtonWithLoading } from ".";

describe("ButtonWithLoading", () => {
  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(
      <ButtonWithLoading onClick={function() {}}>
        Give Me More
      </ButtonWithLoading>,
      div
    );
    ReactDOM.unmountComponentAtNode(div);
  });

  test("has a valid snapshot", () => {
    const component = renderer.create(
      <ButtonWithLoading onClick={function() {}}>
        Give Me More
      </ButtonWithLoading>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
