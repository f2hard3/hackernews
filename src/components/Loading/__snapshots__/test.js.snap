// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonWithLoading has a valid snapshot 1`] = `
<button
  className=""
  onClick={[Function]}
  type="button"
>
  Give Me More
</button>
`;
